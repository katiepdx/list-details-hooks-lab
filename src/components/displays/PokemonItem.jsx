import React from 'react'
import propTypes from 'prop-types'
import './PokemonItem.css'
export default function PokemonItem({ pokemon, url_image }) {
  return (
    <div key={pokemon}>
      <p>Pokemon: {pokemon}</p>
      <img src={url_image} />
    </div>
  )
}

PokemonItem.propTypes = {
  pokemon: propTypes.string.isRequired,
  url_image: propTypes.string.isRequired
}
