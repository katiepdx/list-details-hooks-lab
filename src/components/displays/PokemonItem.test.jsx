// tests that a pokemon item renders to the screen using test data

import React from 'react'
import { render } from '@testing-library/react'
import PokemonItem from './PokemonItem'

describe('Tests PokemonItem component', () => {
  it('should render a list item to the page', () => {
    const { asFragment } = render(<PokemonItem
      name="venusaur"
      image="http://assets.pokemon.com/assets/cms2/img/pokedex/full/003.png"
    />)

    expect(asFragment()).toMatchSnapshot();
  })
})
