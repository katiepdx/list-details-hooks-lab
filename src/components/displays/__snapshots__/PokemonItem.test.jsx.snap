// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests PokemonItem component should render a list item to the page 1`] = `
<DocumentFragment>
  <div>
    <p>
      Pokemon: venusaur
    </p>
    <img
      src="http://assets.pokemon.com/assets/cms2/img/pokedex/full/003.png"
    />
  </div>
</DocumentFragment>
`;
